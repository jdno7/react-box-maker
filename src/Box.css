.Box {
    position: relative;
}
.Box-btn {
    position: absolute;
    right: 0px;
    opacity: .75;
}