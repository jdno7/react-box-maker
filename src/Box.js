import './Box.css'

const Box = ({color, height, width, remove, id}) => {

    return (
        <div data-testid={`BG-${color}`} className="Box" style={{
            backgroundColor: `${color}`,
             height: `${height}`,
             width: `${width}`
        }}>
            <button onClick={() => remove(color)}className="Box-btn" data-testid={color}>
                X
            </button>
        </div>
    )
}

export default Box;