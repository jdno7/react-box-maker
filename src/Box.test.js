import { render, screen } from '@testing-library/react';
import Box from './Box';

test('renders Box Component', () => {
  render(<Box />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
});

test('Snapshot Test', () => {
    const {asFragment} = render(<Box/>)
    expect(asFragment()).toMatchSnapshot()
})

