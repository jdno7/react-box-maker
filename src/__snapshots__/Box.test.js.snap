// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test 1`] = `
<DocumentFragment>
  <div
    class="Box"
    data-testid="BG-undefined"
  >
    <button
      class="Box-btn"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
